src/Microsoft.VisualStudio.Editors/SettingsDesigner/TypePickerDialog.vb

```diff
@@ -197,11 +197,17 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
                     Node.RemoveDummyNode()
                     Using mtsrv As New VSDesigner.MultiTargetService(_vsHierarchy, _projectItemid, False)
                         If (mtsrv IsNot Nothing) Then
-                            Dim typs As Type() = mtsrv.GetSupportedTypes(Node.Text, AddressOf GetAssemblyCallback)
-                            For Each availableType As Type In typs
-                                'TypeTextBox.AutoCompleteCustomSource.Add(availableType.FullName)
-                                If availableType.FullName.Contains(".") AndAlso SettingTypeValidator.IsValidSettingType(mtsrv.GetRuntimeType(availableType)) Then
-                                    _typeTreeView.AddTypeNode(Node, availableType.FullName)
+                            Dim availableTypes As Type() = mtsrv.GetSupportedTypes(Node.Text, AddressOf GetAssemblyCallback)
+                            For Each availableType As Type In availableTypes
+
+                                If availableType.FullName.Contains(".") Then
+
+                                    ' NOTE that GetRuntimeType returns null when there's a failure resolving the type
+                                    Dim runtimeType = mtsrv.GetRuntimeType(availableType)
+                                    If runtimeType IsNot Nothing And SettingTypeValidator.IsValidSettingType(runtimeType) Then
```

Comment:
Should this And be an AndAlso? Without that the behaviour is unchanged isn't it (ie, SettingTypeValidator.IsValidSettingType can still be passed a null).